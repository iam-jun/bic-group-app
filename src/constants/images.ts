const HomeNight = require('../../assets/images/HomeNight.png');
const DiscoveryLogo = require('../../assets/images/img_discovery_logo.png');
const BannerActivePlan = require('../../assets/images/BannerActivePlan.png');

export default {
  HomeNight,
  DiscoveryLogo,
  BannerActivePlan,
};
