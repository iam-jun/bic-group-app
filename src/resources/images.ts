const images = {
  logo: require('../../assets/images/logo.png'),
  logo_bein: require('../../assets/images/bein_logo.png'),
  img_auth_forgot_password_complete: require('../../assets/images/img_auth_forgot_password_complete.png'),
  img_cover_default: require('../../assets/images/cover_default.png'),
  img_groups: require('../../assets/images/groups.png'),
  img_user_avatar_default: require('../../assets/images/user_avatar_default.png'),
  img_group_avatar_default: require('../../assets/images/group_avatar_default.png'),
  img_delete: require('../../assets/images/img_delete.png'),
  img_search_empty: require('../../assets/images/search_empty.png'),
  img_right_col: require('../../assets/images/img_right_col.png'),
};

export default images;
