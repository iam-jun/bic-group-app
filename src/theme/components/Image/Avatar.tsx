import React, {useState} from 'react';
import {StyleSheet} from 'react-native';
import useAuth from '~/hooks/auth';
import {Avatar as RNPAvatar} from 'react-native-paper';

export interface Props {
  uri?: string | undefined | React.ReactNode;
  size?: keyof typeof sizes;
  [x: string]: any;
}

const sizes = {
  small: 24,
  base: 40,
  large: 50,
  big: 60,
};

const Avatar: React.FC<Props> = ({
  style,
  size = 'base',
  user,
  useMyProfile,
  numberOfChars = 2,
  ...props
}) => {
  const [_user, setUser] = useState(user);
  const {user: myProfile} = useAuth();

  React.useEffect(() => {
    if (useMyProfile) setUser(myProfile);
    else setUser(user);
  }, [user, myProfile]);

  if (_user?.avatar) {
    return (
      <RNPAvatar.Image
        {...props}
        style={[styles.container, style]}
        size={sizes[size]}
        source={{uri: _user.avatar}}
      />
    );
  }

  return (
    <RNPAvatar.Text
      style={[styles.container, style]}
      size={sizes[size]}
      label={(_user?.fullName || _user?.name || '').substr(0, 2).toUpperCase()}
    />
  );
};

const styles = StyleSheet.create({
  container: {},
});

Avatar.defaultProps = {
  size: 'base',
};

export default Avatar;
