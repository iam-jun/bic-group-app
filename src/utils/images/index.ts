import { Linking, Platform } from 'react-native';
import i18next from 'i18next';
import RNFetchBlob from 'rn-fetch-blob';
import Clipboard from '@react-native-clipboard/clipboard';
import { PastedFile } from 'react-native-paste-image-input';
import { CameraRoll } from '@react-native-camera-roll/camera-roll';
import { IFilePicked, IToastMessage } from '~/interfaces/common';
import { checkPermission, PermissionTypes } from '../permission';
import showToast from '~/store/helper/showToast';
import showAlert from '~/store/helper/showAlert';
import { ToastType } from '~/baseComponents/Toast/BaseToast';
import { formatBytes } from '../formatter';
import { AppConfig } from '~/configs';

const downloadImageiOS = (photo: any) => {
  const onPermissionGranted = async () => {
    await CameraRoll.save(photo?.url, { type: 'photo' });
    const toastMessage: IToastMessage = {
      content: i18next.t('common:text_downloaded'),
    };

    showToast(toastMessage);
  };

  const onPermissionRefused = () => {
    showAlert({
      title: i18next.t('error:alert_title'),
      content: i18next.t('common:permission_add_photo_blocked'),
      cancelBtn: true,
      confirmLabel: i18next.t('common:text_go_to_settings'),
      onConfirm: () => {
        Linking.openSettings();
      },
    });
  };

  const onCallback = (isGranted: boolean) => {
    if (isGranted) {
      onPermissionGranted();
    } else {
      onPermissionRefused();
    }
  };

  checkPermission(PermissionTypes.AddPhoto, onCallback, false);
};

const downloadImageAndroid = (photo: any) => {
  const onPermissionGranted = async () => {
    try {
      const path = `${RNFetchBlob.fs.dirs.PictureDir}/${photo?.name}`;
      await RNFetchBlob.config({ path }).fetch('GET', photo?.url);
      await RNFetchBlob.fs.scanFile([{ path }]);
      const toastMessage: IToastMessage = {
        content: i18next.t('common:text_downloaded'),
      };

      showToast(toastMessage);
    } catch (error) {
      console.error('downloadImageAndroid error: ', error);
    }
  };

  const onPermissionRefused = () => {
    showAlert({
      title: i18next.t('error:alert_title'),
      content: i18next.t('common:permission_add_photo_blocked'),
      cancelBtn: true,
      confirmLabel: i18next.t('common:text_go_to_settings'),
      onConfirm: () => Linking.openSettings(),
    });
  };

  const onCallback = (isGranted: boolean) => {
    if (isGranted) {
      onPermissionGranted();
    } else {
      onPermissionRefused();
    }
  };

  checkPermission(PermissionTypes.AddPhoto, onCallback, false);
};

export const downloadImage = (photo: any) => {
  if (!photo) return;

  if (Platform.OS === 'ios') {
    downloadImageiOS(photo);
  } else {
    downloadImageAndroid(photo);
  }
};

export const copyImageFromUrl = async (url: string) => {
  if (!url) return;

  try {
    const response = await RNFetchBlob.config({ fileCache: true }).fetch('GET', url);
    const imagePath = response.path();
    const base64Data = await response.readFile('base64');

    RNFetchBlob.fs.unlink(imagePath);
    Clipboard.setImage(base64Data);

    const toastMessage: IToastMessage = {
      content: i18next.t('common:copied'),
    };
    showToast(toastMessage);
  } catch (error) {
    console.error('copyImageFromUrl error:', error);
  }
};

export const getImagePastedFromClipboard = (files: any[]) => {
  const imgs = files.filter((file) => !file.error);
  if (imgs && imgs.length > 0) {
    const img: PastedFile = {
      fileName: imgs[0].fileName,
      type: imgs[0].type,
      fileSize: imgs[0].fileSize,
      uri: imgs[0].uri,
    };
    return img;
  }
  return null;
};

export const checkFileSelected = (file: IFilePicked) => {
  if (Platform.OS === 'ios') {
    if (file?.sourceURL?.includes('GIF') || file?.sourceURL?.includes('WEBP')) {
      const error = i18next.t('common:error:file:file_type_not_support');
      showToast({ content: error, type: ToastType.ERROR });
      return false;
    }
  }

  if (file?.size > AppConfig.groupImageMaxSize) {
    const error = i18next.t('common:error:file:file_exceed_limit').replace('{n}', formatBytes(AppConfig.groupImageMaxSize, 0));
    showToast({ content: error, type: ToastType.ERROR });
    return false;
  }

  return true;
};
