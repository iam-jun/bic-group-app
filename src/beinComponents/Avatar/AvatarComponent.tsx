import React from 'react';
import {
  ImageStyle, StyleProp, StyleSheet, View, ViewStyle,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Icon from '~/beinComponents/Icon';
import Image, { ImageProps } from '~/beinComponents/Image';
import { IconType } from '~/resources/icons';
import dimension from '~/theme/dimension';

import spacing from '~/theme/spacing';
import Text from '../Text';

export type AvatarType =
  | 'tiny'
  | 'small'
  | 'smallAlt'
  | 'medium'
  | 'large'
  | 'largeAlt'
  | 'ultraSuperLarge';

export type AvatarStatus = 'online';

export interface AvatarProps extends ImageProps {
  style?: StyleProp<ImageStyle>;
  variant?: AvatarType;
  status?: AvatarStatus;
  actionIcon?: IconType;
  badge?: any;
  badgeBottom?: boolean;
  isRounded?: boolean;
  cache?: boolean;
  showBorder?: boolean;
  badgeCheck?: boolean;
  badgeCheckSize?: number;
  iconCheckSize?: number;
  counter?: number;
  onPressAction?: () => void;
}

const AvatarComponent: React.FC<AvatarProps> = ({
  variant = 'medium',
  style,
  source,
  status,
  actionIcon,
  badge,
  badgeBottom,
  isRounded,
  showBorder,
  badgeCheck,
  badgeCheckSize = 16,
  iconCheckSize = 12,
  counter,
  onPressAction,
  ...props
}: AvatarProps) => {
  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = creatStyle(theme);

  const avatarSize = dimension?.avatarSizes[variant] || 24;
  const avatarContainerStyle: StyleProp<ViewStyle> = styles[variant];
  let avatarStyle: StyleProp<ImageStyle> = styles[variant];
  const borderStyle = showBorder ? styles.border : {};

  if (isRounded) {
    avatarStyle = StyleSheet.flatten([
      avatarStyle,
      { borderRadius: avatarSize / 2 },
    ]);
  }

  const renderAction = () => {
    if (!onPressAction) {
      return null;
    }

    const actionContainerSize = 16;
    const actionIconSize = 12;

    return (
      <View
        testID="avatar.action_icon"
        style={{
          position: 'absolute',
          top: -(spacing?.margin.tiny || 4),
          right: -(spacing?.margin.tiny || 4),
          width: actionContainerSize,
          height: actionContainerSize,
          borderRadius: actionContainerSize / 2,
          backgroundColor: colors.white,
          justifyContent: 'center',
          alignItems: 'center',
        }}
      >
        {actionIcon && (
          <Icon
            testID="avatar.action_icon.button"
            size={actionIconSize}
            icon={actionIcon}
            onPress={onPressAction}
          />
        )}
      </View>
    );
  };

  const renderStatus = () => {
    if (!status) {
      return null;
    }

    // todo update in specific case
    const statusSize = avatarSize / 3;
    const statusColor = colors.success;

    return (
      <View
        testID="avatar.status"
        style={{
          position: 'absolute',
          bottom: 0,
          right: 0,
          width: statusSize,
          height: statusSize,
          borderRadius: statusSize / 2,
          backgroundColor: statusColor,
          borderWidth: 1,
          borderColor: colors.white,
        }}
      />
    );
  };

  const renderBadge = () => {
    if (!badge) {
      return null;
    }

    // todo update in specific case
    const badgeContainerSize = avatarSize / 2;
    const badgeSize = badgeContainerSize * 0.45;

    const absTop = -(spacing?.margin.small || 8);
    const absBottom = -(spacing?.margin.tiny || 4);
    const absRight = -(spacing?.margin.small || 8);

    return (
      <View
        testID="avatar.badge"
        style={{
          position: 'absolute',
          top: badgeBottom ? undefined : absTop,
          bottom: badgeBottom ? absBottom : undefined,
          right: absRight,
          width: badgeContainerSize,
          height: badgeContainerSize,
          borderRadius: badgeContainerSize / 2,
          backgroundColor: colors.white,
          justifyContent: 'center',
          alignItems: 'center',
          shadowOffset: { width: 0, height: 1 },
          shadowColor: '#000',
          shadowOpacity: 0.1,
          shadowRadius: 1,
          elevation: 2,
        }}
      >
        <Icon size={badgeSize} icon={badge} />
      </View>
    );
  };

  const renderBadgeCheck = () => {
    if (!badgeCheck) {
      return null;
    }
    return (
      <View
        testID="avatar.badge_check"
        style={{
          position: 'absolute',
          bottom: 0,
          right: 0,
          width: badgeCheckSize,
          height: badgeCheckSize,
          borderRadius: badgeCheckSize / 2,
          backgroundColor: colors.success,
          justifyContent: 'center',
          alignItems: 'center',
          borderWidth: 1,
          borderColor: colors.white,
          shadowOffset: { width: 0, height: 1 },
          shadowColor: '#000',
          shadowOpacity: 0.1,
          shadowRadius: 1,
          elevation: 2,
        }}
      >
        <Icon size={iconCheckSize} icon="Check" tintColor={colors.neutral80} />
      </View>
    );
  };

  return (
    <View
      testID="avatar"
      style={StyleSheet.flatten([avatarContainerStyle, style])}
    >
      <View
        testID="avatar_container"
        style={StyleSheet.flatten([
          avatarStyle,
          source ? {} : { backgroundColor: colors.gray40 },
        ])}
      >
        <Image
          testID="avatar.image"
          style={[avatarStyle, borderStyle]}
          source={source}
          {...props}
        />
        {counter ? (
          <View style={[avatarStyle, styles.textStyle]}>
            <Text.BodyS color={theme.colors.white}>
              +
              {counter}
            </Text.BodyS>
          </View>
        ) : null}
        {renderStatus()}
        {renderAction()}
        {renderBadge()}
        {renderBadgeCheck()}
      </View>
    </View>
  );
};

const creatStyle = (theme: ExtendedTheme) => {
  const { colors } = theme;
  return StyleSheet.create({
    container: {},
    tiny: {
      width: dimension?.avatarSizes?.tiny,
      height: dimension?.avatarSizes?.tiny,
      borderRadius: (dimension?.avatarSizes?.tiny || 16) / 2,
    },
    small: {
      width: dimension?.avatarSizes?.small,
      height: dimension?.avatarSizes?.small,
      borderRadius: spacing?.borderRadius.small,
    },
    smallAlt: {
      width: dimension?.avatarSizes?.smallAlt,
      height: dimension?.avatarSizes?.smallAlt,
      borderRadius: spacing?.borderRadius.large,
    },
    medium: {
      width: dimension?.avatarSizes?.medium,
      height: dimension?.avatarSizes?.medium,
      borderRadius: spacing?.borderRadius.small,
    },
    large: {
      width: dimension?.avatarSizes?.large,
      height: dimension?.avatarSizes?.large,
      borderRadius: spacing?.borderRadius.small,
    },
    largeAlt: {
      width: dimension?.avatarSizes?.largeAlt,
      height: dimension?.avatarSizes?.largeAlt,
      borderRadius: (spacing?.borderRadius.small || 6) + 2,
    },
    ultraSuperLarge: {
      width: dimension?.avatarSizes?.ultraSuperLarge,
      height: dimension?.avatarSizes?.ultraSuperLarge,
      borderRadius: (spacing?.borderRadius.small || 6) + 2,
    },
    border: {
      borderWidth: 4,
      borderColor: colors.white,
    },
    textStyle: {
      backgroundColor: colors.transparent1,
      position: 'absolute',
      justifyContent: 'center',
      alignItems: 'center',
    },
  });
};

export default AvatarComponent;
