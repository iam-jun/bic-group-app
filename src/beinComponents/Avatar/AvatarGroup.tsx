import React from 'react';
import {StyleProp, StyleSheet, View, ViewStyle} from 'react-native';
import {useTheme} from 'react-native-paper';
import AvatarComponent, {
  AvatarProps,
  AvatarType,
} from '~/beinComponents/Avatar/AvatarComponent';
import Image from '~/beinComponents/Image';
import Text from '~/beinComponents/Text';
import images from '~/resources/images';
import dimension from '~/theme/dimension';
import {fontFamilies} from '~/theme/fonts';
import {ITheme} from '~/theme/interfaces';
import spacing from '~/theme/spacing';

export interface AvatarGroupProps extends AvatarProps {
  variant?: AvatarType;
  style?: StyleProp<ViewStyle>;
  source?: any;
  totalMember?: number;
}

const AvatarGroup = ({
  variant = 'medium',
  style,
  source,
  totalMember,
  ...props
}: AvatarGroupProps) => {
  const theme = useTheme() as ITheme;
  const {colors} = theme;

  const containerSize = dimension?.avatarSizes?.[variant] || 36;
  const border = 2;
  const itemContainerSize = (containerSize + border * 2) / 2;
  const itemSize = (containerSize - border * 2) / 2;

  const totalMemberFontSizes: {[x: string]: any} = {
    small: 6,
    medium: 8,
    large: 10,
    ultraLarge: 14,
  };
  const totalMemberFontSize = totalMemberFontSizes[variant] || 8;

  const renderItem = (
    index: number,
    source?: string,
    style?: StyleProp<ViewStyle>,
  ) => {
    return (
      <View
        testID={`avatar_group.item_${index}`}
        style={StyleSheet.flatten([
          {
            width: itemContainerSize,
            height: itemContainerSize,
            borderWidth: border,
            borderColor: colors.background,
            borderRadius: spacing.borderRadius.small,
            backgroundColor: colors.background,
            overflow: 'hidden',
          },
          style,
        ])}>
        <Image
          {...props}
          style={{
            width: itemSize,
            height: itemSize,
            backgroundColor: colors.primary1,
          }}
          source={source}
        />
      </View>
    );
  };

  const renderItems = () => {
    if (!Array.isArray(source))
      return (
        <AvatarComponent
          {...props}
          source={source}
          placeholderSource={images.img_group_avatar_default}
        />
      );

    switch (source?.length) {
      case 1:
        return renderItem(0, source?.[0]);
      case 2:
        return (
          <View testID="avatar_group.group_2" style={{flexDirection: 'row'}}>
            {renderItem(0, source?.[0])}
            {renderItem(1, source?.[1], {marginLeft: -6})}
          </View>
        );
      case 3:
        return (
          <View testID="avatar_group.group_3">
            <View style={{flexDirection: 'row'}}>
              {renderItem(0, source?.[0])}
              {renderItem(1, source?.[1], {marginLeft: -6})}
            </View>
            <View
              style={{
                flexDirection: 'row',
                justifyContent: 'center',
                marginTop: -8,
              }}>
              {renderItem(2, source?.[2])}
            </View>
          </View>
        );
      case 4:
        return (
          <View testID="avatar_group.group_4">
            <View style={{flexDirection: 'row'}}>
              {renderItem(0, source?.[0])}
              {renderItem(1, source?.[1], {marginLeft: -6})}
            </View>
            <View
              style={{
                flexDirection: 'row',
                justifyContent: 'center',
                marginTop: -8,
              }}>
              {renderItem(2, source?.[2])}
              {renderItem(3, source?.[3], {marginLeft: -6})}
            </View>
          </View>
        );
      default:
        return (
          <View testID="avatar_group.group_4_plus">
            <View style={{flexDirection: 'row'}}>
              {renderItem(0, source?.[0])}
              {renderItem(1, source?.[1], {marginLeft: -6})}
            </View>
            <View
              style={{
                flexDirection: 'row',
                justifyContent: 'center',
                marginTop: -8,
              }}>
              {renderItem(2, source?.[2])}
              <View
                style={{
                  width: itemContainerSize,
                  height: itemContainerSize,
                  backgroundColor: colors.background,
                  borderRadius: 6,
                  marginLeft: -6,
                  justifyContent: 'center',
                  alignItems: 'center',
                }}>
                {totalMember ? (
                  <Text
                    testID="avatar_group.total_member"
                    style={{
                      fontFamily: fontFamilies.OpenSans,
                      marginTop: 2,
                      fontSize: totalMemberFontSize,
                    }}>
                    {`+${Math.min(99, totalMember)}`}
                  </Text>
                ) : (
                  renderItem(3, source?.[3])
                )}
              </View>
            </View>
          </View>
        );
    }
  };

  return (
    <View
      testID="avatar_group"
      style={StyleSheet.flatten([
        {
          width: containerSize,
          height: containerSize,
          backgroundColor: colors.primary1,
          borderRadius: spacing.borderRadius.small,
          justifyContent: 'center',
          alignItems: 'center',
          padding: 1,
        },
        style,
      ])}>
      {renderItems()}
    </View>
  );
};

export default AvatarGroup;
