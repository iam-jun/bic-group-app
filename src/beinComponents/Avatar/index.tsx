import React from 'react';
import AvatarComponent, {
  AvatarProps,
} from '~/beinComponents/Avatar/AvatarComponent';

const Tiny: React.FC<AvatarProps> = (props: AvatarProps) => (
  <AvatarComponent variant={'tiny'} {...props} />
);
const Small: React.FC<AvatarProps> = (props: AvatarProps) => (
  <AvatarComponent variant={'small'} {...props} />
);
const Medium: React.FC<AvatarProps> = (props: AvatarProps) => (
  <AvatarComponent variant={'medium'} {...props} />
);
const Large: React.FC<AvatarProps> = (props: AvatarProps) => (
  <AvatarComponent variant={'large'} {...props} />
);
const UltraLarge: React.FC<AvatarProps> = (props: AvatarProps) => (
  <AvatarComponent variant={'ultraLarge'} {...props} />
);

const Avatar = Object.assign(AvatarComponent, {
  Tiny,
  Small,
  Medium,
  Large,
  UltraLarge,
});

export default Avatar;
