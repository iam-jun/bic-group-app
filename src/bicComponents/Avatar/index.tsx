import React from 'react';
import AvatarComponent, {
  AvatarProps,
} from '~/bicComponents/Avatar/AvatarComponent';

const Tiny: React.FC<AvatarProps> = (props: AvatarProps) => (
  <AvatarComponent variant="tiny" {...props} />
);
const XSmall: React.FC<AvatarProps> = (props: AvatarProps) => (
  <AvatarComponent variant="xSmall" {...props} />
);
const Small: React.FC<AvatarProps> = (props: AvatarProps) => (
  <AvatarComponent variant="small" {...props} />
);
const Base: React.FC<AvatarProps> = (props: AvatarProps) => (
  <AvatarComponent variant="base" {...props} />
);
const Medium: React.FC<AvatarProps> = (props: AvatarProps) => (
  <AvatarComponent variant="medium" {...props} />
);
const Large: React.FC<AvatarProps> = (props: AvatarProps) => (
  <AvatarComponent variant="large" {...props} />
);
const XLarge: React.FC<AvatarProps> = (props: AvatarProps) => (
  <AvatarComponent variant="xLarge" {...props} />
);

const Avatar = Object.assign(
  AvatarComponent, {
    Tiny,
    XSmall,
    Small,
    Base,
    Medium,
    Large,
    XLarge,
  },
);

export default Avatar;
