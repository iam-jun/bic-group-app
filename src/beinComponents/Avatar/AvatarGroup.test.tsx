import * as React from 'react';
import {render, cleanup} from '@testing-library/react-native';

import Avatar from '~/beinComponents/Avatar';
import images from '~/resources/images';

afterEach(cleanup);

describe('Avatar Group component', () => {
  const avatarLink = images.logo_bein;
  const listAvatar2 = [
    'https://cdn.dribbble.com/users/183984/screenshots/2562247/pokemon.jpg',
    'https://cdn.dribbble.com/users/183984/screenshots/2569843/pokemon_3.jpg',
  ];
  const listAvatar3 = [
    'https://cdn.dribbble.com/users/183984/screenshots/2562247/pokemon.jpg',
    'https://cdn.dribbble.com/users/183984/screenshots/2569843/pokemon_3.jpg',
    'https://cdn.dribbble.com/users/183984/screenshots/2565088/pokemon_2.jpg',
  ];
  const listAvatar4 = [
    'https://cdn.dribbble.com/users/183984/screenshots/2562247/pokemon.jpg',
    'https://cdn.dribbble.com/users/183984/screenshots/2569843/pokemon_3.jpg',
    'https://cdn.dribbble.com/users/183984/screenshots/2565088/pokemon_2.jpg',
    'https://cdn.dribbble.com/users/183984/screenshots/2574264/pokemon_4.jpg',
  ];

  const listAvatar5 = [
    'https://cdn.dribbble.com/users/183984/screenshots/2562247/pokemon.jpg',
    'https://cdn.dribbble.com/users/183984/screenshots/2569843/pokemon_3.jpg',
    'https://cdn.dribbble.com/users/183984/screenshots/2565088/pokemon_2.jpg',
    'https://cdn.dribbble.com/users/183984/screenshots/2582592/pokemon_5.jpg',
    'https://cdn.dribbble.com/users/183984/screenshots/2574264/pokemon_4.jpg',
  ];

  it(`renders correctly`, () => {
    const rendered = render(<Avatar.Group />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it(`renders correctly group with no member`, () => {
    const {getByTestId} = render(<Avatar.Group />);
    const avatarComponent = getByTestId('avatar_container');
    expect(avatarComponent).toBeDefined();
  });

  it(`renders correctly avatar when call Avatar.Group`, () => {
    const {getByTestId} = render(<Avatar.Group source={avatarLink} />);
    const avatarComponent = getByTestId('avatar.image');
    expect(avatarComponent).toBeDefined();
  });

  it(`renders correctly group with 1 member`, () => {
    const {getByTestId} = render(<Avatar.Group source={[avatarLink]} />);
    const avatarComponent = getByTestId('avatar_group.item_0');
    expect(avatarComponent).toBeDefined();
  });

  it(`renders correctly group with 2 member`, () => {
    const {getByTestId} = render(<Avatar.Group source={listAvatar2} />);
    const groupComponent = getByTestId('avatar_group.group_2');
    expect(groupComponent).toBeDefined();
  });

  it(`renders correctly group with 3 member`, () => {
    const {getByTestId} = render(<Avatar.Group source={listAvatar3} />);
    const groupComponent = getByTestId('avatar_group.group_3');
    expect(groupComponent).toBeDefined();
  });

  it(`renders correctly group with 4 member`, () => {
    const {getByTestId} = render(<Avatar.Group source={listAvatar4} />);
    const groupComponent = getByTestId('avatar_group.group_4');
    expect(groupComponent).toBeDefined();
  });

  it(`renders correctly group with 4+ member with totalMember`, () => {
    const {getByTestId} = render(
      <Avatar.Group source={listAvatar5} totalMember={100} />,
    );
    const groupComponent = getByTestId('avatar_group.group_4_plus');
    expect(groupComponent).toBeDefined();
    const totalMemberComponent = getByTestId('avatar_group.total_member');
    expect(totalMemberComponent).toBeDefined();
  });

  it(`renders correctly group with 4+ member without totalMember`, () => {
    const {getByTestId} = render(<Avatar.Group source={listAvatar5} />);
    const groupComponent = getByTestId('avatar_group.group_4_plus');
    expect(groupComponent).toBeDefined();
    const fourthItemComponent = getByTestId('avatar_group.item_3');
    expect(fourthItemComponent).toBeDefined();
  });
});
