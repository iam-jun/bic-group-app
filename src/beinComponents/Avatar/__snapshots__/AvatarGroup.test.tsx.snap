// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Group component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F8F5FF",
      "borderRadius": 6,
      "height": 40,
      "justifyContent": "center",
      "padding": 1,
      "width": 40,
    }
  }
  testID="avatar_group"
>
  <View
    style={
      Object {
        "borderRadius": 6,
        "height": 40,
        "width": 40,
      }
    }
    testID="avatar"
  >
    <View
      style={
        Object {
          "backgroundColor": "#B2BDCD",
          "borderRadius": 6,
          "height": 40,
          "width": 40,
        }
      }
      testID="avatar_container"
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "position": "relative",
          }
        }
      >
        <View
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f9f9f9",
                "borderRadius": 6,
                "height": 40,
                "justifyContent": "center",
                "width": 40,
              }
            }
            testID="Image__placeholder"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 6,
                "height": 40,
                "width": 40,
              },
            ]
          }
        >
          <FastImageView
            onFastImageError={[Function]}
            onFastImageLoadEnd={[Function]}
            resizeMode="cover"
            source={
              Object {
                "testUri": "../../../assets/images/group_avatar_default.png",
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
            testID="avatar.image"
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
